nautobot_golden_config/views.py

```diff
@@ -78,13 +79,39 @@ def __init__(self, *args, **kwargs):
         self.config_details = None
         self.action_template_name = None
 
-    @property
-    def dynamic_group_queryset(self):
-        """Return queryset of DynamicGroups associated with all GoldenConfigSettings."""
-        golden_config_device_queryset = Device.objects.none()
+    def filter_queryset(self, queryset):
+        """Add a warning message when GoldenConfig Table is out of sync."""
+        queryset = super().filter_queryset(queryset)
+        # Only adding a message when no filters are applied
+        if self.filter_params:
+            return queryset
+
+        out_of_sync_message = mark_safe(
+            "The GoldenConfig Table is missing entries and/or has additional entries "
+            "based on what is defined for the DynamicGroups in GoldenConfigSettings. "
+            "Run this job to rectify this issue."  # TODO 2.x: Add job_url
+        )
+
+        gc_dynamic_group_device_queryset = Device.objects.none()
         for setting in models.GoldenConfigSetting.objects.all():
-            golden_config_device_queryset = golden_config_device_queryset | setting.dynamic_group.members
-        return golden_config_device_queryset & self.queryset.distinct()
+            gc_dynamic_group_device_queryset = gc_dynamic_group_device_queryset | setting.dynamic_group.members
+
+        gc_dynamic_group_device_queryset = gc_dynamic_group_device_queryset.distinct()
+        gc_dynamic_group_device_queryset_count = gc_dynamic_group_device_queryset.count()
+        gc_queryset = models.GoldenConfig.objects.all()
+        gc_queryset_count = gc_queryset.count()
+
+        # Only validating Devices are the same for count < 100, otherwise same count means they are in sync.
+        if gc_dynamic_group_device_queryset_count < 100:
+            gc_device_queryset = Device.objects.filter(pk__in=[gc.device.pk for gc in gc_queryset])
+            dg_to_gc_difference = gc_dynamic_group_device_queryset.difference(gc_device_queryset)
+            gc_to_dg_difference = gc_device_queryset.difference(gc_dynamic_group_device_queryset)
+            if dg_to_gc_difference or gc_to_dg_difference:
+                messages.warning(self.request, message=out_of_sync_message)
+        elif gc_dynamic_group_device_queryset_count != gc_queryset_count:
+            messages.warning(self.request, message=out_of_sync_message)
```

Comment:
Pretty sure you can get away with this, or something like it (untested):

```suggestion
      gc_dynamic_group_device_ids = set(gc_dynamic_group_device_queryset.distinct().values_list("pk", flat=True))
      gc_device_ids = set(models.GoldenConfig.objects.values_list("device__pk", flat=True))
      
      if gc_dynamic_group_device_ids != gc_device_ids:
          messages.warning(self.request, message=out_of_sync_message)
```

Reply:
ya, I was trying to figure out how to just get the Device's from GoldenConfig yesterday, but I guess I was thinking about the problem wrong, and grabbing just the PKs is doable

Reply:
also, I don't think `distinct` is needed when using `|`